python/private/python.bzl

```
@@ -355,21 +438,146 @@ interpreter to create pyc files for the standard library, but, because they are
+                  "granularity (e.g. '3.12.1') is not supported.",
+        ),
+    },
+)
+
+_override = tag_class(
+    doc = """Tag class used to override defaults and behaviour of the module extension.
+
+:::{versionadded} 0.36.0
+:::
+""",
+    attrs = {
+        "available_python_versions": attr.string_list(
+            mandatory = False,
+            doc = "The list of available python tool versions to use. Must be in `X.Y.Z` format.",
+        ),
+        "base_url": attr.string(
+            mandatory = False,
+            doc = "The base URL to be used when downloading toolchains.",
+            default = DEFAULT_RELEASE_BASE_URL,
+        ),
+
+        # Internal attributes that are only usable from `rules_python`
+        "register_all_versions": attr.bool(default = False, doc = "`rules_python` **internal** use only!"),
+    },
+)
+
+_single_version_override = tag_class(
+    doc = """Override single python version URLs and patches for all platforms.
+
+:::{note}
+This will replace any existing configuration for the given python version.
+
+If you would like to modify the configuration for a specific `(version,
+platform), please use the {rule}`python.single_version_platform_override` tag
+class.
+:::
+
+:::{versionadded} 0.36.0
+:::
+""",
+    attrs = {
+        "patch_strip": attr.int(
+            mandatory = False,
+            doc = "Same as the --strip argument of Unix patch.",
+            default = 0,
+        ),
+        "patches": attr.label_list(
+            mandatory = False,
+            doc = "A list of labels pointing to patch files to apply for the interpreter repository. They are applied in the list order and are applied before any platform-specific patches are applied.",
+        ),
+        "sha256": attr.string_dict(
+            mandatory = True,
+            doc = "The python platform to sha256 dict. See {attr}`python.single_version_platform_override.platform` for allowed key values.",
+        ),
+        "strip_prefix": attr.string(
+            mandatory = False,
+            doc = "The 'strip_prefix' for the archive, defaults to 'python'.",
+            default = "python",
+        ),
+        "urls": attr.string_list(
+            mandatory = True,
+            doc = "The URL template to fetch releases for this Python version. See {attr}`python.single_version_platform_override.urls` for documentation.",
+        ),
+        "version": attr.string(
+            mandatory = True,
+            doc = "The python version to override URLs for. Must be in `X.Y.Z` format.",
+        ),
+    },
+)
+
+_single_version_platform_override = tag_class(
+    doc = """Override single python version for a single existing platform.
+
+If the `(version, platform)` is new, we will add it to the existing versions and will
+use the same `url` template.
+
+:::{note}
+If you would like to add or remove platforms to a single python version toolchain
+configuration, please use {rule}`python.single_version_override`.
```

Yeah, `{rule}` doesn't work to link to a tag class. Try `{obj}`, which is a more generic lookup. I don't think there's a xref role specific to tag classes.

```suggestion
configuration, please use {obj}`python.single_version_override`.
```